more_modbus/RegisterMapping.cpp

```diff
@@ -424,6 +430,28 @@ const std::chrono::milliseconds& RegisterMapping::getRepeatedWrite() const
     return m_repeatedWrite;
 }
 
+void RegisterMapping::setRepeatedWrite(const std::chrono::milliseconds& repeatedWrite)
+{
+    // Check if the repeated write needs to be added now into the device's list
+    if (m_repeatedWrite.count() == 0 && repeatedWrite.count() > 0 &&
+        (m_group != nullptr && m_group->getDevice() != nullptr))
+    {
+        m_group->m_device->addRewritable(shared_from_this());
```

Comment:
Is there really no other way of doing this?
We have three circular references device<->group<->mapping
Use some sort of observer pattern at least if these mappings have to be updated this way
But you really need to break these dependencies as these objects can never be deleted.